src/graph/visitor/test/ExtractGroupSuiteVisitorTest.cpp

```diff
@@ -0,0 +1,380 @@
+        std::cout << "g2.groupItems[i]=" << g2.groupItems[i]->toString() << std::endl;
+        return false;
+      }
+    }
+    return true;
+  }
+};
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestConstantExpression) {
+  ExtractGroupSuiteVisitor visitor;
+  constantExpr(1)->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(constantExpr(1));
+  expect.groupItems.push_back(constantExpr(1));
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestUnaryExpression1) {
+  auto* e = notExpr(constantExpr(1));
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(e);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestUnaryExpression2) {
+  auto* agg = aggExpr("collect", constantExpr(1), false);
+  auto* e = notExpr(agg);
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  // expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(agg);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestTypeCastingExpression) {
+  auto* e = castExpr(Type::BOOL, constantExpr(0));
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(e);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestLabelExpression) {
+  auto* e = labelExpr("aaaaa");
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(e);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestLabelAttributeExpression) {
+  auto* e = laExpr("abc", "10");
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(e);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestArithmeticExpression1) {
+  auto* left = constantExpr(0);
+  auto* right = constantExpr(1);
+  auto* e = addExpr(left, right);
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(e);
+  expect.groupItems.push_back(e);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
+
+TEST_F(ExtractGroupSuiteVisitorTest, TestArithmeticExpression2) {
+  auto* left = constantExpr(0);
+  auto* right = aggExpr("count", constantExpr(1), false);
+  auto* e = minusExpr(left, right);
+
+  ExtractGroupSuiteVisitor visitor;
+  e->accept(&visitor);
+
+  GroupSuite expect;
+  expect.groupKeys.push_back(left);
+  expect.groupItems.push_back(left);
+  expect.groupItems.push_back(right);
+
+  ASSERT_EQ(true, check(visitor.groupSuite(), expect));
+}
```

Comment:
That's incorrect.
`v.age+count(v.name)` key: v.age items: v.age,count(v.name)
`1+count(v.name)` item: count(v.name)

Reply:
You mean, that constantExpr doesn't need to put groupkeys and groupItems?